dht/table.go

```diff
@@ -93,12 +107,14 @@ func (table *InMemTable) AddPeer(peerID id.Signatory, peerAddr wire.Address) {
 
 	// Insert into the sorted address list based on its XOR distance from our
 	// own address.
-	i := sort.Search(len(table.sorted), func(i int) bool {
-		return table.isCloser(peerID, table.sorted[i])
-	})
-	table.sorted = append(table.sorted, id.Signatory{})
-	copy(table.sorted[i+1:], table.sorted[i:])
-	table.sorted[i] = peerID
+	if !ok {
+		i := sort.Search(len(table.sorted), func(i int) bool {
+			return table.isCloser(peerID, table.sorted[i])
+		})
+		table.sorted = append(table.sorted, id.Signatory{})
+		copy(table.sorted[i+1:], table.sorted[i:])
+		table.sorted[i] = peerID
```

Comment:
You need to add a test for `AddPeer` that verifies this behaviour works as intended (e.g. that the peer can be updated, but its position in the sorted list does not change).